### ISSUE TITLE
[cannot execute] When pnpm build, occurred %1 is not a valid Win32 application

### ISSUE THREAD

**Issue by kooriookami** [pnpm/pnpm#5638]
### pnpm version:
7.16.0

### Code to reproduce the issue:
https://github.com/kooriookami/electron-builder-pnpm

### Expected behavior:
Run `pnpm build` no errors.

### Actual behavior:
![image](https://user-images.githubusercontent.com/38392315/202068367-dbcccf83-95e9-48b4-890d-2ef98a3ee298.png)

### Additional information:

Use sqlite3 in dependencies.

 - `node -v` prints: v16.16.0
 - Windows, macOS, or Linux?: Win11 x64

**Comment by kooriookami** [comment]
When I use npm to build, it works well.
![image](https://user-images.githubusercontent.com/38392315/202071021-7bf1537f-869a-43ec-b529-6b4e6437aa79.png)

**Comment by v9u** [comment]
The same questions, demo:https://github.com/v9u/sqlite_test

**Comment by v9u** [comment]
> The same questions, demo:https://github.com/v9u/sqlite_test

 fix way,open `pnpm.cjs` file, replace `#!/usr/bin/env node` with `#!node`,from https://github.com/electron-userland/electron-builder/issues/6933#issuecomment-1213438889

**Comment by kooriookami** [comment]
> The same questions, demo:https://github.com/v9u/sqlite_test

Thanks! You solved my question. I can build electron now. But it steel has an error. I will use npm util pnpm solve it.
![image](https://user-images.githubusercontent.com/38392315/204078081-2e1c341e-f930-4f71-9bb7-aa1b58062eba.png)

**Comment by v9u** [comment]
> > The same questions, demo:https://github.com/v9u/sqlite_test
> 
> Thanks! You solved my question. I can build electron now. But it still has an error. I will use npm until pnpm solve it. ![image](https://user-images.githubusercontent.com/38392315/204078081-2e1c341e-f930-4f71-9bb7-aa1b58062eba.png)

Because `pnpm build` not include @mapbox/node-pre-gyp , I don't know why

**Comment by v9u** [comment]
> > The same questions, demo:https://github.com/v9u/sqlite_test
> 
> Thanks! You solved my question. I can build electron now. But it still has an error. I will use npm until pnpm solve it. ![image](https://user-images.githubusercontent.com/38392315/204078081-2e1c341e-f930-4f71-9bb7-aa1b58062eba.png)

I solved the problem,look: https://github.com/v9u/sqlite_test

**Comment by kooriookami** [comment]
We may wait for pnpm's reply.

**Comment by v9u** [comment]
> 

Yes,but i think it is electron-builder's question,look this:https://www.electron.build/index.html#note-for-pnpm

**Comment by AiHaibara** [comment]
> > The same questions, demo:https://github.com/v9u/sqlite_test
> 
> fix way,open `pnpm.cjs` file, replace `#!/usr/bin/env node` with `#!node`,from [electron-userland/electron-builder#6933 (comment)](https://github.com/electron-userland/electron-builder/issues/6933#issuecomment-1213438889)

  ⨯ cannot execute  cause=fork/exec C:\Users\10200362\AppData\Roaming\npm\node_modules\pnpm\bin\pnpm.cjs: %1 is not a valid Win32 application.
                    command='C:\Users\10200362\AppData\Roaming\npm\node_modules\pnpm\bin\pnpm.cjs' rebuild bufferutil@4.0.7 utf-8-validate@5.0.10

this can be fixed in my windows by edit 'AppData\Roaming\npm\node_modules\pnpm\bin\pnpm.cjs', but how to let other users auto fix this, all users need change like this?

**Comment by thoriqdharmawan** [comment]
> The same questions, demo:https://github.com/v9u/sqlite_test

thanks, this works for me 🙌

**Comment by Buzhifanji** [comment]
Two years have passed and this problem still exists
